### PR TITLE
[Android] Customizable dialog for shared mode

### PR DESCRIPTION
**Pull request by lincsoon**
Implement the interface for customizing how the dialog would be
displayed when initializing Crosswalk Project runtime.

BUG=XWALK-5211

### CODE REVIEW

**Comment by lincsoon** [comment]
@wuhengzhi @xzhan96 @fujunwei @axinging Please help to review

File: runtime/android/core/src/org/xwalk/core/XWalkActivity.java [hunk 1]
```diff
@@ -118,33 +118,63 @@
...
     protected void onXWalkFailed() {
         finish();
     }
 
     /**
-     * Return true if Crosswalk Project runtime is ready, false otherwise.
+     * Set up the dialog manager so that you can customize the dialog to be dislplayed when
+     * initializing Crosswalk Project runtime. This method must be called within onCreate(). Once
+     * onResume() is invoked, a default dialog manager will be set up and you won't be able to
+     * changed it.
```
Comment (wuhengzhi) [inline runtime/android/core/src/org/xwalk/core/XWalkActivity.java]: changed -> change?

File: runtime/android/core/src/org/xwalk/core/XWalkDialogManager.java [hunk 2]
```diff
@@ -164,24 +361,61 @@ private AlertDialog buildAlertDialog() {
...
+
+    private void showDialog(final AlertDialog dialog, final ArrayList<ButtonAction> actions) {
+        dialog.setOnShowListener(new OnShowListener() {
+            @Override
+            public void onShow(DialogInterface d) {
+                for (ButtonAction action : actions) {
+                    Button button = dialog.getButton(action.mWhich);
+                    if (button == null) {
+                        if (action.mMandatory) {
+                            throw new RuntimeException("Button " + action.mWhich + " is mandatory");
```
Comment (wuhengzhi) [inline runtime/android/core/src/org/xwalk/core/XWalkDialogManager.java]: `mWhich`'s value is `BUTTON_POSITIVE` or `BUTTON_NEGATIVE`, they are integer, is it make sense to user?

File: runtime/android/core/src/org/xwalk/core/XWalkUpdater.java [hunk 3]
```diff
@@ -302,8 +300,14 @@ public XWalkUpdater(XWalkUpdateListener listener, Activity activity) {
         mDialogManager = new XWalkDialogManager(activity);
     }
 
-    // This constructor is for XWalkActivityDelegate
-    XWalkUpdater(XWalkUpdateListener listener, Activity activity,
+    /**
+     * Create XWalkUpdater for single activity
```
Comment (wuhengzhi) [inline runtime/android/core/src/org/xwalk/core/XWalkUpdater.java]: In some comments, the period is missing.

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java [hunk 4]
```diff
@@ -103,77 +97,95 @@
...
+ * {@link XWalkActivity} or {@link XWalkInitializer}. For example:</p>
  *
  * <pre>
- *   import android.app.Activity;
- *   import android.os.Bundle;
+ * import android.content.Intent;
+ * import android.os.Bundle;
+ * import android.webkit.ValueCallback;
+ * import org.xwalk.core.XWalkActivity;
+ * import org.xwalk.core.XWalkViewInternal;
```
Comment (wuhengzhi) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: Alpha sort for these imports is needed to change?

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java [hunk 5]
```diff
@@ -103,77 +97,95 @@
...
- *       }
+ *         // Until onXWalkReady() is invoked, you should do nothing with the
+ *         // embedding API except the following:
+ *         // 1. Instantiate the XWalkView object
+ *         // 2. Call XWalkPreferences.setValue()
+ *         // 3. Call mXWalkView.setXXClient(), e.g., setUIClient
+ *         // 4. Call mXWalkView.setXXListener(), e.g., setDownloadListener
+ *         // 5. Call mXWalkView.addJavascriptInterface()
+ *         setContentView(R.layout.activity_main);
+ *         mXWalkView = (XWalkView) findViewById(R.id.xwalkview);
```
Comment (wuhengzhi) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: XWalkView -> XWalkViewInternal?

File: tools/reflection_generator/reflection_generator.py [hunk 6]
```diff
@@ -35,18 +37,16 @@
...
-    'XWalkResourceClientInternal',
-    'XWalkPreferencesInternal',
-    'XWalkNavigationItemInternal',
-    'XWalkNavigationHistoryInternal',
-    'XWalkNativeExtensionLoaderInternal',
     'XWalkJavascriptResultHandlerInternal',
     'XWalkJavascriptResultInternal',
+    'XWalkNativeExtensionLoaderInternal',
+    'XWalkNavigationItemInternal',
+    'XWalkNavigationHistoryInternal',
```
Comment (wuhengzhi) [inline tools/reflection_generator/reflection_generator.py]: XWalkNavigationHistoryInternal
XWalkNavigationItemInternal

File: runtime/android/core/src/org/xwalk/core/XWalkDialogManager.java (hunk 2 above)
Comment (lincsoon) [inline runtime/android/core/src/org/xwalk/core/XWalkDialogManager.java]: It's just for internal use.

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java (hunk 4 above)
Comment (lincsoon) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: It's a sample. So I think it doesn't matter

**Pull request merged**